src/cr/cube/cube.py

```diff
@@ -380,7 +408,7 @@ def unweighted_counts(self):
     @lazyproperty
     def valid_counts(self):
         """ndarray of valid counts, valid elements only."""
-        valid_counts = self._measure.valid_counts
+        valid_counts = self._base_measure.valid_counts
         if valid_counts.any():
             return valid_counts[self._valid_idxs]
         return np.empty(0)
```

Comment:
This use of an empty array to signal "no value" is improper, because it is an "in-band" value, i.e. it is the same type as a valid value, and who's to say that one day (or even already) an empty array might be a valid return value.

When signaling the failure case of an optional value, use an out-of-band value, which is exactly what `None` is. It's also easier and faster to check, with `if valid_counts is None:`.

I'm pretty sure these lines are going away anyway, but the principle applies broadly, not just in this instance.